lib/cadence/workflow/poller.rb

```diff
@@ -29,10 +30,10 @@ def wait
 
       private
 
-      attr_reader :domain, :task_list, :client, :workflow_lookup, :middleware
+      attr_reader :domain, :task_list, :client, :workflow_lookup, :middleware, :options
 
       def client
-        @client ||= Cadence::Client.generate
+        @client ||= Cadence::Client.generate(options)
```

Comment:
Depending on where this gets called from it gets a different set of options (maybe with ttl, maybe with thread pool size)

Reply:
Yes, the plan is to initialize it in the worker and just pass the reference